### PR TITLE
Prefab - Gnomish Warship

### PR DESCRIPTION
**Pull request by vocalpocal**
## About the PR
This is a contest entry for https://forum.ss13.co/showthread.php?tid=16992
Summary:

- Size: 13x19
- Probability: 25
- Enemies: 2 (Maneaters)
- Merchant: David the Gnome
- Other: Two garden gnomes which water plants
- Equipment: Three hydroponic trays which spawn with strange seeds in them - otherwise regular trays, four arable soil trays, and some green clothes

## The gnomish warship
image

The cockpit is fiercely defended by two very hungry maneaters. The engine room has been renovated into a laboratory for alien plant research. The ship is manned by two gnome critters, who keep the garden pretty.  In the cockpit you can find the ship's merchant captain, David the Space Gnome!

## David the Space Gnome
image
image

David sells many useful botany supplies, including fertilizers and strange seeds. He is quite stingy, so his wares don't come cheap! He himself buys honey (because it is tasty), human and monkey meat (to feed the maneaters), but not synth meat.

David the Spacegnome has two secret items, but he wont sell them to just anyone! In order to unlock his most valuable secret shop items, you have to show him a very high-class botany produce. He is not interested in herbs or other crap you might produce, he wants to see a quality snack!

## Gnome critters
image

These helpful fellows water plants when they need it most. They also clear trays from dead plants, and very rarely plant seeds to empty hydroponic trays. Huggable. Unfortunately, they are mute.

## Why's this needed? 
This is a contest entry. Gnomes are cute. 
The merchant sells many useful (yet expensive) tools for botanists to counteract some of the unintended side effects of piss nerfing -- botany actually suffered a lot from it and countless strats were invalidated when saltpetre output dropped by over 90%. You can get saltpetre through other means (such as QM, fabricating more GardenGears) but emptying dozens of tiny bottles into bigger containers makes it very tiresome. David sells saltpetre in larger bottles, making the process a bit easier (if you have the money). 
The 50000 credit splicing module enables very interesting strategies which have not been feasible before. When installed to your Plantmaster Mk3, it improves its minimum splice chance from 0% to 30%. If you somehow get past the maneaters, and have the money for the module, then you have almost unlimited potential in your splices --- but failures are still possible.

## Changelog

```changelog
(u)Vocalpocal
(*)Added a gnomish warship prefab. Go say hi to David the Space Gnome, who sells hydroponics supplies!
```

### CODE REVIEW

**Comment by Crackaduck** [comment]
Amazing, only complaint is that there's no cargo pad at the to more easily get stuff to it.

**Comment by Zonespace27** [comment]
Reiterating on what was said on the forums, the LUCA seems incredibly overpowered, considering it _gives a permanent 30% splice chance, regardless of genome_, which I sense that the "easy" ways of getting 50k will be quickly rushed, should the prefab appear on the mining map PDA app.

**Review by Tarmunora — CHANGES_REQUESTED** [review]
Remove the trader and its commodities and this is good to go I think

**Comment by vocalpocal** [comment]
> Reiterating on what was said on the forums, the LUCA seems incredibly overpowered, considering it gives a permanent, minimum 30% splice chance, regardless of genome, which I sense that the "easy" ways of getting 50k will be quickly rushed, should the prefab appear on the mining map PDA app.

The 30% chance matters only on very distant species, it does not increase splicing chances of species that are already quite similar. My gut feeling says that this is could make some weird mixes possible at 20 minute mark (and not any earlier) while it was possible in 50-70 minutes before. Although 20 is still very slow when compared to some other antag stuff in the game, I was thinking of ways to make it even slower. In order to unlock LUCA, you must deliver a Jumbo-sized produce to prove that you are a strong enough botanist to handle such alien tech! This would easily add 10 or 15 minutes to even the most rigorous speedrunners... And they would also have to know how to do botany in the first place.

> Remove the trader and its commodities and this is good to go I think

But without the trader there is no loot in the ship and it loses its purpose. Are you sure there are no compromises you are ready to take here?

**Comment by Azrun** [comment]
>
> But without the trader there is no loot in the ship and it loses its purpose.

Preferably it would be worth visiting because it was interesting, not just because there is loot to gather.  I think it seems goofy and fun. 😃

**Comment by Tarmunora** [comment]
> >
> > But without the trader there is no loot in the ship and it loses its purpose.
> 
> Preferably it would be worth visiting because it was interesting, not just because there is loot to gather.  I think it seems goofy and fun. 😃 

Additionally, there are still the garden gnomes which seem to automatically seek out and water crops

**Comment by vocalpocal** [comment]
> Additionally, there are still the garden gnomes which seem to automatically seek out and water crops

They are technically nothing new for botany gameplay. They are just Botanical Misters with legs.

**Comment by unfunnyperson** [comment]
Removing the trader would take the fun out of the prefab in my personal opinion, just remove the splicing module or whatever and its fine.

**Review by virvatuli-gh — CHANGES_REQUESTED** [review]
Where did you get the sound files, are they licensed correctly? Additionally, they're unnecessarily stereo and could easily be compressed by quite a bit. Did you read the audio guidelines?

**Comment by vocalpocal** [comment]
Sounds are from the same source as the original gnome chuckle -- they are just cut parts from the original video. For the compression part I just did what the original gnomechuckle.ogg did and applied the same to these new sound files.

**Comment by Sovexe** [comment]
We would need a proper and compatible license from the author to add their sounds to the repo unless this all open domain 😕

**Comment by Azrun** [comment]
Perhaps this is a silly question... What makes this a warship?

**Comment by vocalpocal** [comment]
> Perhaps this is a silly question... What makes this a warship?

I was planning to add turrets outside the ship which fire seething tomatoes... Needs some work but I might be able to push it before the deadline. The name just stayed that way, even though its current version does not have guns. The ship is shapes like a gnome head though.

File: code/obj/critter/gnome.dm [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+	angertext = "chants unintelligible gibberish at"
+	flags = FPRINT | CONDUCT | USEDELAY | TABLEPASS | FLUID_SUBMERGE | FLUID_SUBMERGE
+
+	var/plant_check = 20
+	var/working_range = 1
+	var/plant_plan
+	var/obj/machinery/plantpot/work_target
+	var/work_bore = 40
+
+	ai_think()
```
[the remark this answers is not in the document]
Comment (vocalpocal) [reply code/obj/critter/gnome.dm:25]: Wait, I think the critter parent already does this on its own, unless I am completely mistaken.

**Pull request closed without merging**